test/plugins/wasm_bpf/wasm_bpf.cpp

```diff
@@ -0,0 +1,578 @@
+    if (const auto *Module = Plugin->findModule("wasm_bpf"sv)) {
+      return Module->create().release();
+    }
+  }
+  return nullptr;
+}
+
+std::filesystem::path getAssertsPath() {
+  std::filesystem::path thisFile(__FILE__);
+  return thisFile.parent_path() / "asserts";
+}
+void fillMemContent(WasmEdge::Runtime::Instance::MemoryInstance &memInst,
+                    uint32_t offset, uint32_t count, char chr = 0) noexcept {
+  std::fill_n(memInst.getPointer<char *>(offset), count, chr);
+}
+
+void fillMemContent(WasmEdge::Runtime::Instance::MemoryInstance &memInst,
+                    uint32_t offset, const std::vector<char> &data) noexcept {
+  char *buf = memInst.getPointer<char *>(offset);
+  std::copy(data.begin(), data.end(), buf);
+}
+
+} // namespace
+
+TEST(WasmBpfTest, Module) {
+  auto module = dynamic_cast<WasmEdge::Host::WasmBpfModule *>(createModule());
+  EXPECT_NE(module, nullptr);
+  // Test whether functions are exported
+  EXPECT_EQ(module->getFuncExportNum(), 6U);
+  EXPECT_NE(module->findFuncExports("wasm_load_bpf_object"), nullptr);
+  EXPECT_NE(module->findFuncExports("wasm_close_bpf_object"), nullptr);
+  EXPECT_NE(module->findFuncExports("wasm_attach_bpf_program"), nullptr);
+  EXPECT_NE(module->findFuncExports("wasm_bpf_buffer_poll"), nullptr);
+  EXPECT_NE(module->findFuncExports("wasm_bpf_map_fd_by_name"), nullptr);
+  EXPECT_NE(module->findFuncExports("wasm_bpf_map_operate"), nullptr);
+
+  delete module;
+}
+
+static const size_t TASK_COMM_LEN = 16;
+static const size_t MAX_FILENAME_LEN = 127;
+struct event {
+  int pid;
+  int ppid;
+  unsigned exit_code;
+  unsigned long long duration_ns;
+  char comm[TASK_COMM_LEN];
+  char filename[MAX_FILENAME_LEN];
+  char exit_event;
+};
+
+class PollCallbackFunction
+    : public WasmEdge::Runtime::HostFunction<PollCallbackFunction> {
+public:
+  PollCallbackFunction() {}
+  WasmEdge::Expect<int32_t> body(const WasmEdge::Runtime::CallingFrame &Frame,
+                                 uint32_t __attribute__((unused)) ctx,
+                                 uint32_t data, uint32_t data_sz) {
+    using namespace std;
+    using WasmEdge::unlikely;
+    auto *memory = Frame.getMemoryByIndex(0);
+    if (unlikely(!memory)) {
+      return WasmEdge::Unexpect(WasmEdge::ErrCode::Value::HostFuncError);
+    }
+    if (data_sz < static_cast<uint32_t>(sizeof(event))) {
+      return WasmEdge::Unexpect(WasmEdge::ErrCode::Value::HostFuncError);
+    }
+    const event *dataPtr = memory->getPointer<const event *>(data, 1);
+    if (unlikely(!dataPtr)) {
+      return WasmEdge::Unexpect(WasmEdge::ErrCode::Value::HostFuncError);
+    }
+    auto nowTime = chrono::system_clock::to_time_t(chrono::system_clock::now());
+    tm nowTimeRepr;
+    localtime_r(&nowTime, &nowTimeRepr);
+    if (dataPtr->exit_event == 1) {
+      cout.setf(ios::left);
+      cout << std::put_time(&nowTimeRepr, "%H:%M:%S") << " EXIT " << setw(16)
```

Comment:
I think there is no need to print out the data, just check whether the result matches the format and is correctly send to user space?

For example, output a test data in kernel with ring buffer:

```c
e->exit_event = false;
e->pid = 1;
e->ppid = 2;
```
There is no need to use the real data.

Reply:
I write a new test which uses fixed data to test. Just remove the the test if you want